### ISSUE TITLE
png8 causing issues

### ISSUE THREAD

**Issue by waynegraham** [performant-software/Neatline#330]
In OpenLayers we're requesting the `image/png8` format. However, not every WMS service offers this optimized version. This is specifically a problem with the NYPL and Harvard Map Warper projects. We should update this to just request the `image/png` format.

**Comment by gusunavarro** [comment]
Ok Wayne, I hope that you can help me
Thanks
gus

**Comment by gusunavarro** [comment]
Wayne
only can see these
gus

![layers](https://cloud.githubusercontent.com/assets/924400/4112879/a87a7536-323c-11e4-8bdd-2fecaf29e1da.png)

**Comment by waynegraham** [comment]
@gusunavarro Are you setting the WMS for the entire exhibit, or just the "Huela personal..." record?

**Comment by waynegraham** [comment]
@gusunavarro Oh, and what base map are you using?

**Comment by gusunavarro** [comment]
Wayne
Thanks very much for your answer. Yes I use wms for the entire exhibit. 
but I only can see these map and not my list of spatial layers.But for see need use a lot of zoom + -. Maybe you can see  www.  http://www.koluel.org/neatline/show/huelgas-personal-ferrocarril   I use base spatial layers OSM.
![spatial](https://cloud.githubusercontent.com/assets/924400/4123076/bbaa7212-32ca-11e4-95c6-a3e1e14341f2.png)
![spatial2](https://cloud.githubusercontent.com/assets/924400/4123081/c3019d2e-32ca-11e4-8324-766fef38a61a.png)

**Comment by waynegraham** [comment]
I think may be a different issue than the png8 issue; does this sound sound like what is going on?

When you set a base WMS layer from MapWarper (Harvard or NYPL), spatial layers on top of that base layer don't appear. 

Does this change if you remove this map as the base layer and use something like the Open Street Map layer and then add this in as a new record?

**Comment by gusunavarro** [comment]
here in mapwarper with OSM
http://warp.worldmap.harvard.edu/maps/3433#Preview_Map_tab

and here with
http://worldmap.harvard.edu/maps/koluel

**Comment by gusunavarro** [comment]
Does this change if you remove this map as the base layer and use something like the Open Street Map layer and then add this in as a new record?

Do you have any example?

**Comment by gusunavarro** [comment]
Wayne , sorry for insistence, but here in my university we need show these.
I created and example but I don't see the layers

for example
these is my new record
http://www.koluel.org/neatline/show/lituania

![wms](https://cloud.githubusercontent.com/assets/924400/4133923/3bedbb06-3365-11e4-952b-24e130b89a05.png)

**Comment by waynegraham** [comment]
Change the WMS layer name to 3433

![neatline editor lituania](https://cloud.githubusercontent.com/assets/45000/4134381/e493b2f0-336b-11e4-9c63-23a862328db7.png)

**Comment by gusunavarro** [comment]
Wayne
Thanks very much for your answer. I add the same that your image . but I don't see the image.
![style3](https://cloud.githubusercontent.com/assets/924400/4136118/df0fa4c8-337b-11e4-951b-2781957ad952.png)s

maybe you can see in my site, what is the problem?
I add to my site an account
wayne
123456

**Comment by gusunavarro** [comment]
www.koluel.org/admin

**Comment by gusunavarro** [comment]
Wayne
Today  I try as you said
I add 3433 to WMS layer name but I don't have luck.
![layers6](https://cloud.githubusercontent.com/assets/924400/4144467/263092d0-33e1-11e4-8875-08f8a5b01b8b.png)

sorry for my questions.
gus

**Comment by waynegraham** [comment]
What is the full WMS address? It should have the 3433 at the end as well...

**Comment by gusunavarro** [comment]
http://warp.worldmap.harvard.edu/maps/wms/3433

**Comment by gusunavarro** [comment]
Dear Wayne

Do you have any idea of the problem?
Thanks
gus

**Comment by gusunavarro** [comment]
from Harvard

Yes we have not publicized WMS capabilities out of WorldMap so it is not in the documentation.  Here it is:

Notes:

Warper is working for bringing WMS layers either into WorldMap or Neatline.  You should not need to bring layers from Warper into WorldMap to create WMSs from them.

WorldMap is working for bringing WMS layers into Neatline.

Getting WMSs to work in Neatline this has only worked for me by adding a Record to the neatline map, but not in the Exhibit configuration section where a WMS should be definable as a base map.

Example:

Here is a Map in worldmap http://worldmap.harvard.edu/maps/238

The WMS capabilities request for layers in this map is:

http://worldmap.harvard.edu/capabilities/map/238

Here is one layer in that Map (238):

http://worldmap.harvard.edu/data/geonode:railroads_7q6

A layer name in this Map is:  railroads_7q6

Here is this railroad layer brought into Neatline:

http://neatline.worldmap.harvard.edu/neatline/show/ben-test

Let me know if this helps.

Ben

Ben Lewis

**Comment by davidmcclure** [comment]
Hey @gusunavarro,

Two things are going on here. First, you need to use the plaintext identifier for the WMS layer that's getting served out of WorldMap, which, in this case, is just `image`. I don't know too much about WorldMap, so I'm not sure if this is always the default, or if it could be configured somewhere - I found it just by inspecting the outgoing WMS requests from http://worldmap.harvard.edu/maps/koluel.

But, this still won't work because it seems that the WMS server behind WorldMap is throwing 500's when Neatline asks for tiles in `image/png8` format, instead of the more standard `image/png`. This was a change that I made about a month ago (`png8` files are smaller than regular `png`'s, which results in a nice little speed boost). But, since some WMS servers apparently choke on this, we'll have to roll this back. I just pushed up a branch called [`bug/png8`](https://github.com/scholarslab/Neatline/tree/bug/png8) that fixes this, and I confirmed that it works with your layer. So, as a stop-gap, just clone off that branch, and change the WMS fields to be:

WMS Address: `http://warp.worldmap.harvard.edu/maps/wms/3433`
WMS Layers: `image`

We'll get the `png8` fix merged into master ASAP and cut off a point release. In the meantime, let me know if you hit problems.

Best,
David

**Comment by waynegraham** [comment]
@davidmcclure now we have two branches with the same fix :) bug/png8 and feature/map_warperfix. Just need to prune those when we merge this.

@gusunavarro I'm not sure you have the correct branch checked out on your server...Try downloading this version on your server: 

``` shell
$ cd path/to/omeka/plugins
$ wget https://github.com/scholarslab/Neatline/archive/bug/png8.zip
$ unzip png8.zip
$ mv Neatline /tmp && mv Neatline-bug-png8 Neatline
```

I've also been in touch with Ben about getting some instructions on MapWarper too...

Wayne

**Comment by davidmcclure** [comment]
Doh, sorry, I'm juggling like 10 support threads right now and can't keep them all straight in my head. I bet the `feature/map_warperfix` branch isn't working for @gusunavarro because the change in the source hasn't been compiled into the production payloads that get tracked in the repo (this is done with `grunt freeze`). `bug/png8` should work after just cloning/downloading the repo, without any build steps.

**Comment by gusunavarro** [comment]
here
http://www.koluel.org/neatline/show/santa-cruz#records/1

**Comment by gusunavarro** [comment]
http://www.koluel.org/neatline/show/santa-cruz

**Comment by davidmcclure** [comment]
Cool! Now, you'll probably want to bump up the "Fill Opacity" setting on the record, which will make the layer more opaque. Also, in the long run, you might want to try georectifying a higher-resolution version of the image, which will make it look less pixelated when you zoom in close.

**Comment by gusunavarro** [comment]
thanks very much to all

**Comment by mkane2** [comment]
Hi @waynegraham and @davidmcclure -- sorry if this is resurrecting an old thread, but I've installed the the bug/png8 branch to use a WorldMap WMS in Neatline, and still have no success in getting theWMS layer to show up in Neatline.  My rectified map shows up in WorldMap itself just fine from the export WMS url provided in warper: http://worldmap.harvard.edu/maps/5764.  The map itself in warper http://warp.worldmap.harvard.edu/maps/3744 is set to public.

I've tried WMS Address: http://warp.worldmap.harvard.edu/maps/wms/3744
and WMS Layers: image as described above, with no luck.

Thanks for your help!

M

**Comment by waynegraham** [comment]
@mkane2 I believe this has been merged in to the develop branch; could you give that one a try?

**Comment by gusunavarro** [comment]
when  I have these problem I use these version
 cd path/to/omeka/plugins
$ wget https://github.com/scholarslab/Neatline/archive/bug/png8.zip
$ unzip png8.zip
$ mv Neatline /tmp && mv Neatline-bug-png8 Neatline

**Comment by mkane2** [comment]
@waynegraham and @gusunavarro thanks, but still no luck with either the develop branch and this branch.  I'm using Stamen Toner and OpenStreetMap; the record that the WMS is in shows up, but the rectified map does not.  

Currently I've got the develop branch up, with WMS Address: http://warp.worldmap.harvard.edu/maps/wms/3744 WMS Layers: image

![screen shot 2014-10-20 at 5 48 26 pm](https://cloud.githubusercontent.com/assets/9325924/4709857/07de19c4-58a3-11e4-9579-fc786628d403.png)

**Comment by mkane2** [comment]
"Sanborn test" in the screenshot as an overlay is the record with the map that's not showing up, fwiw

**Comment by mkane2** [comment]
I've gotten it working, using the develop branch, but using a different WMS Address.  Instead of the "WMS base URL" http://warp.worldmap.harvard.edu/maps/wms/3744, I used the "Tiles base URL" http://warp.worldmap.harvard.edu/maps/tile/3744/z/x/y.png in MapWarper's Export tab
![screen shot 2014-10-20 at 6 41 35 pm](https://cloud.githubusercontent.com/assets/9325924/4710494/7b75dc6c-58aa-11e4-8bd3-9b82c499ebb1.png)

**Comment by waynegraham** [comment]
@davidmcclure did your `bug/png8` branch get merged in to develop?

**Comment by lindsaytday** [comment]
Hi,

I read through the forum here and on the omeka forum. However, I am still having issues with using georectified maps from WorldMap warper in Neatline. I have figured out that the georectified map I'm using works in Neatline version 2.2.0. However, I have the latest version of Neatline, 2.3.0, and it does not work in my Neatline exhibit. How can I make a change in the latest version? If not, I will uninstall and reinstall the older version.

Thank you!

**Comment by davidmcclure** [comment]
Hey @lindsaytday,

First of all, I'm so sorry for the slow response! The fix for the WorldMap layers actually hasn't been included in an official release yet - this will work when Neatline 2.3.1 is released. Until then, you can get the fix by deploying the `bug/png8` branch, which includes the fix for the problem:

https://github.com/scholarslab/Neatline/tree/bug/png8

It should work to just click the "Download ZIP" link at the bottom of the right column, and then copy the new files into the `Neatline` plugin folder on your server.

Best,
David

**Comment by waynegraham** [comment]
@davidmcclure can we merge that bug branch in to the develop branch? Are there any other blocking issues to release 2.3.1?

**Comment by davidmcclure** [comment]
Yeah, we can just merge that in. I had the thought of making it possible to set a flag in the plugin.ini that would set whether the tiles are requested as `png` / `png8` (since the latter does provide a nice speed boost when the tile server supports it). But that's a nice-to-have, and can be added in the future.

**Comment by gusunavarro** [comment]
Thanks very much. So in the future welcome the version 2.3.1
Gus

2015-01-12 15:25 GMT-03:00 David McClure notifications@github.com:

**Comment by waynegraham** [comment]
This is patched in 2.4.0 merged in to master as of a8fd43f8bbdee822fa35358dae252223c8796bdb

**Comment by lindsaytday** [comment]
Hi, I'm just wondering if anyone has tried to use the David Rumsey georeferencer for georectifying maps and layering them in Neatline? The georeferencer is awesome, but I don't see a way to get the maps into Neatline. 

Thank you!
